test/shape.test.js

```diff
@@ -23,45 +23,48 @@ const expectations = [{
-            const spriter = new SVGSpriter({
-                shape: {
-                    dest: 'svg',
-                    dimension: {
-                        maxWidth: 4000,
-                        maxHeight: 4000
-                    }
+const TEST_SVG = '<svg xmlns="http://www.w3.org/2000/svg"></svg>';
+
+describe('testing shapes', () => {
+    it.each`
+            svg                    | dimension
+            ${expectations[0].svg} | ${expectations[0].result}
+            ${expectations[1].svg} | ${expectations[1].result}
+        `('should call calculateSvgDimensions if the $svg does not contain viewBox or height/width properties ($dimension)', async({
+        svg, dimension
+    }) => {
+        expect.hasAssertions();
+
+        const spriter = new SVGSpriter({
+            shape: {
+                dest: 'svg',
+                dimension: {
+                    maxWidth: 4000,
+                    maxHeight: 4000
                 }
-            });
+            }
+        });
 
-            const svgFilePath = path.join(paths.fixtures, `svg/special/without-dims/${expectation.svg}`);
+        calculateSvgDimensions.mockReturnValueOnce(dimension);
 
-            spriter.add(
-                svgFilePath,
-                expectation.svg,
-                readFileSync(svgFilePath, 'utf-8')
-            );
+        const svgFilePath = path.join(__dirname, `fixture/svg/special/without-dims/${svg}`);
 
-            spriter.compile((error, result) => {
-                try {
-                    assert.ifError(error);
-                    assert.equal(isObject(result), true);
-                    assert.notEqual(typeof result.shapes, 'undefined');
-                    assert.equal(Array.isArray(result.shapes), true);
+        spriter.add(
+            svgFilePath,
+            svg,
+            Buffer.from(TEST_SVG)
+        );
 
-                    const svg = result.shapes[0]._contents.toString();
-                    const dom = new DOMParser().parseFromString(svg, 'text/xml');
+        expect(calculateSvgDimensions).toHaveBeenCalledWith(new DOMParser().parseFromString(`${DEFAULT_XML_DECLARATION}${TEST_SVG}`).toString());
 
-                    assert.equal(dom.documentElement.getAttribute('height'), expectation.result.height.toString());
-                    assert.equal(dom.documentElement.getAttribute('width'), expectation.result.width.toString());
+        const { result } = await spriter.compileAsync();
 
-                    done();
-                } catch (compilationError) {
-                    done(compilationError);
-                }
-            });
-        });
+        expect(result).toBeInstanceOf(Object);
+        expect(result.shapes).toBeInstanceOf(Array);
+
+        const dom = new DOMParser().parseFromString(result.shapes[0]._contents.toString(), 'text/xml');
+
+        expect(dom.documentElement.getAttribute('height')).toStrictEqual(dimension.height.toString());
```

Comment:
is there any way to have jest always use strict assertions like we had `assert.strict`?

Reply:
This is a strange case here, here is the rules https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/prefer-strict-equal.md, https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/prefer-to-be.md.
`toStrictEqual` is used to compare objects (with structure), otherwise it is preferred to use `toBe`.
In this case I think its linting problem (because it can't get type of `getAttribute`) and we can change it to `toBe`, that's better

Reply:
Anyway, preferring strict mode is already covered by eslint

Reply:
I wasn't talking about the `'use strict' directive 😛

I was wondering why you were using `toStrictEqual` since `toEqual` exists too and if we can always enable strict checks like assert has with strict mode. :)